### DISCUSSION TITLE
Spatiotemporal model with GP component

### DISCUSSION THREAD

**Discussion by thomaspinder** [thomaspinder/GPJax#80] [Ideas]
Starting a discussion thread for the idea mentioned by @theorashid in #76 

I'd be really interested – particularly for my own work. I think a spatiotemporal problem would be a nice way to involve that community with your package. A good example dataset would be the [Hungary Chickenpox dataset](https://pytorch-geometric-temporal.readthedocs.io/en/latest/modules/dataset.html#module-torch_geometric_temporal.dataset.chickenpox) . I would like to do this with a graph GP over space kroneckered with a GP over time (see [Fast Hierarchical GPs](https://sethrf.com/files/fast-hierarchical-GPs.pdf)) if that's possible. It's also a Poisson likelihood and we could try and add a covariate in for a larger model. Perhaps when you're more free, we could sit down for a chat about implementing this – many in my spatiotemproal epidemiology crowd would be interested.

**Comment by theorashid** [comment]
The main idea for this would be to do a spatiotemporal regression. The Hungary Chickenpox dataset consists of weekly (over 500) counts of chickenpox cases for counties (20). Here are the ingredients we would need:
- Poisson likelihood for count data. The effects below model the log(rate).
- Space. It is possible to use a 2D continuous spatial setting using the centroid of each county, but I would prefer to use a GP graph kernel where vertices are counties and edges define neighbours (with covariance $K_s$).
- Time. Using a GP (with covariance $K_t$) with seasonal effects, but there's no reason we should't use other time series models like AR(p)
- Kronecker product to define $GP(0, K_s \otimes K_t)$ (see Fast Hierarchical GPs above). If not we could model the latent GPs as independent as $GP(0, K_s) + GP(0, K_t)$.
- Covariate. We could find a continuous covariate and include it as $ + \beta x$

I have no idea whether it's possible to combine a graph kernel and continuous kernel in that way, but given they just define covariance matrices, I can't see why not.

Also, the spatial graph kernel is sparse, especially for spatial situations with loads of units when each unit has very few neighbours. Is there somewhere this might speed things up?
